src/library/itunes/itunesdao.h

```diff
@@ -59,6 +68,11 @@ class ITunesDAO : public DAO {
     QHash<int, QString> m_playlistNameById;
     std::multimap<int, int> m_playlistIdsByParentId;
 
+    // Keeps track of whether the database has been initialized.
+    // In tests the database is not used, so the importer will
+    // only be used to construct a TreeItem.
+    bool m_isDatabaseInitialized = false;
```

Comment:
can you elaborate? How is this related to the tests?

Reply:
Aside from making db calls, the `importPlaylist` method also updates the name-id mapping and the `importPlaylistRelation` updates the parent-child mapping, both of which are needed to construct the `TreeItem` later on. Since we don't need/want the db calls during testing, I'm using this flag currently, which (kind of unintentionally) doubles as a safety check.

There are other approaches I've thought of, such as splitting up db logic and bookkeeping into separate methods, but those make the abstraction leakier and place more of a burden on the caller of the DAO, so I'm not sure whether that would be a better option.